src/txmempool.h

```diff
@@ -23,7 +23,13 @@
 #include <util/result.h>
 #include <util/feefrac.h>
 
+// This works around a bug in Boost <= 1.80.0 when using Clang >=18.
+// See https://github.com/bitcoin/bitcoin/issues/30751.
+#pragma clang diagnostic push
```

Comment:
Should this be in a `#if defined(__clang__)` block?

Reply:
A bit annoying, but yea, looks like we'll need something like that to work around the windows compiler in any case.

Reply:
(depending on what GCC does, over the lifetime of the 28.x branch, at some point, it may turn in and #ifndef MSVC etc)